From fb7fdbd20e8e6ca08b30c679e5e37f78ae40bd5e Mon Sep 17 00:00:00 2001
From: bcoll <bcoll@cloudflare.com>
Date: Thu, 26 Jan 2023 15:30:34 +0000
Subject: [PATCH 09/13] Hide unsupported settings UI

- Show CORS errors in console setting
- Show XHR requests in console setting
- Disable JavaScript setting
- Devices settings tab
- Throttling settings tab
---
 front_end/panels/console/ConsoleView.ts      | 2 --
 front_end/panels/settings/SettingsScreen.ts  | 4 ++++
 front_end/panels/settings/settingsScreen.css | 5 +++++
 3 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/front_end/panels/console/ConsoleView.ts b/front_end/panels/console/ConsoleView.ts
index 3ae38d9fcd..ebad85638f 100644
--- a/front_end/panels/console/ConsoleView.ts
+++ b/front_end/panels/console/ConsoleView.ts
@@ -460,8 +460,6 @@ export class ConsoleView extends UI.Widget.VBox implements
         i18nString(UIStrings.onlyShowMessagesFromTheCurrentContext), i18nString(UIStrings.selectedContextOnly));
     ConsoleView.appendSettingsCheckboxToToolbar(
         settingsToolbarLeft, this.groupSimilarSetting, i18nString(UIStrings.groupSimilarMessagesInConsole));
-    ConsoleView.appendSettingsCheckboxToToolbar(
-        settingsToolbarLeft, this.showCorsErrorsSetting, i18nString(UIStrings.showCorsErrorsInConsole));
 
     const settingsToolbarRight = new UI.Toolbar.Toolbar('', settingsPane.element);
     settingsToolbarRight.makeVertical();
diff --git a/front_end/panels/settings/SettingsScreen.ts b/front_end/panels/settings/SettingsScreen.ts
index 471e3dcfa4..8ad237714b 100644
--- a/front_end/panels/settings/SettingsScreen.ts
+++ b/front_end/panels/settings/SettingsScreen.ts
@@ -377,6 +377,10 @@ export class GenericSettingsTab extends SettingsTab {
     const uiSectionName = Common.Settings.getLocalizedSettingsCategory(category);
     const sectionElement = this.appendSection(uiSectionName);
     for (const settingRegistration of settings) {
+      const name = settingRegistration.settingName;
+      if (name === "monitoringXHREnabled" || name === "consoleShowsCorsErrors" || name === "javaScriptDisabled") {
+        continue;
+      }
       const setting = Common.Settings.Settings.instance().moduleSetting(settingRegistration.settingName);
       const settingControl = UI.SettingsUI.createControlForSetting(setting);
       if (settingControl) {
diff --git a/front_end/panels/settings/settingsScreen.css b/front_end/panels/settings/settingsScreen.css
index 37a79572c0..96dde853c7 100644
--- a/front_end/panels/settings/settingsScreen.css
+++ b/front_end/panels/settings/settingsScreen.css
@@ -243,3 +243,8 @@ label {
     color: HighlightText;
   }
 }
+
+#tab-devices,
+#tab-throttling-conditions {
+  display: none;
+}
-- 
2.39.5 (Apple Git-154)

